web/src/engine/websites/ComicFuz.ts

```diff
@@ -0,0 +1,179 @@
+    }
+
+    private async fetchMangaDetail(titleId: string): Promise<MangaDetailResponse> {
+        const uri = new URL('v1/manga_detail', this.apiUrl);
+        const payload = {
+            deviceInfo: {
+                deviceType: 2,
+            },
+            mangaId: titleId
+        };
+        const request = await this.createPROTORequest(uri, 'ComicFuz.MangaDetailRequest', payload);
+        return FetchProto<MangaDetailResponse>(request, protoTypes, 'ComicFuz.MangaDetailResponse');
+    }
+
+    private async createPROTORequest(uri: URL, messageProtoType: string, payload: unknown): Promise<FetchRequest> {
+        const root = await protobuf.parse(protoTypes, { keepCase: true }).root;
+        const messageType = root.lookupType(messageProtoType);
+        const message = messageType.encode(payload);
+        return new FetchRequest(uri.href, {
+            body: message.finish(),
+            method: 'POST'
+        });
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const uri = new URL('v1/mangas_by_day_of_week', this.apiUrl);
+        const payload = {
+            deviceInfo: {
+                deviceType: 2
+            },
+            dayOfWeek: 0
+        };
+        const request = await this.createPROTORequest(uri, 'ComicFuz.MangasByDayOfWeekRequest', payload);
+        const data = await FetchProto<MangasByDayOfWeekResponse>(request, protoTypes, 'ComicFuz.MangasByDayOfWeekResponse');
+        return data.mangas.map(manga => new Manga(this, provider, manga.id.toString(), manga.title));
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const data = await this.fetchMangaDetail(manga.Identifier);
+        const chapters: Chapter[] = [];
+        data.chapters.forEach(chapterGroup => {
+            chapterGroup.chapters.forEach(chapter => {
+                chapters.push(new Chapter(this, manga, chapter.id.toString(), chapter.title.trim()));
+            });
+        });
+        return chapters;
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const uri = new URL('v1/manga_viewer', this.apiUrl);
+        const payload = {
+            deviceInfo: {
+                deviceType: 2
+            },
+            chapterId: chapter.Identifier,
+            useTicket: false,
+            consumePoint: {
+                event: 0,
+                paid: 0
+            }
+        };
+        const request = await this.createPROTORequest(uri, 'ComicFuz.MangaViewerRequest', payload);
+        let data: MangaViewerResponse = undefined;
+        try {
+            data = await FetchProto<MangaViewerResponse>(request, protoTypes, 'ComicFuz.MangaViewerResponse');
+        } catch (error) {
+            throw new Exception(R.Plugin_Common_Chapter_UnavailableError);
+        }
+        return data.pages
+            .filter(page => page.image && page.image.imageUrl)
```

Comment:
May be simplified e.g., `.filter(page => page.image?.imageUrl && page.image.isExtraPage != true)`